src/gsCore/gsPiecewiseFunction.h

```diff
@@ -133,6 +133,8 @@ class gsPiecewiseFunction : public gsFunctionSet<T>
 
     index_t size() const {return m_funcs.size();}
 
+    index_t nPieces() const {return m_funcs.size();}
```

Comment:
size_t

Reply:
https://github.com/gismo/gismo/blob/69e6b51d76b2ee3c48135a5aad2c8804d1d9d4ea/src/gsCore/gsFunctionSet.h#L556-L561

index_t gsPolyField<>::nPieces should overwrite index_t gsFunctionSet<>::nPieces()

Should I change index_t gsFunctionSet<>::nPieces() as well? That will cause a lot of problems. Otherwise polymorphism will not work for index_t gsPolyField<>::nPieces

Reply:
OK, let it as it is, but please solve the `index -1` at line 1725 as you would do it. I'll then update the type when I do it for the rest of the code.